### PR TITLE
Implement gnss7 reading. Determined default address to be 0x42.

### LINKED ISSUE #19: Implement I2c and discover port used by gnss7.

### LINKED ISSUE #20: Read data from Gnss7

### PR DESCRIPTION
**Pull request by stackunderfl0w**
Also update slightly incorrect chip name.
Closes #19
Closes #20

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [target.'cfg(all(target_arch = "arm", target_os = "none"))']
 # replace STM32F103C8 with your chip as listed in `probe-rs chip list`
-runner = "probe-rs run --chip STM32F103C8"
+runner = "probe-rs run --chip STM32F103CBT6"
```
Comment (jake-g00dwin) [inline .cargo/config.toml:3]: Is there a reason to change the stm32f103C8 --> stm32f103CBT6?
The 64k --> 128k difference means we would have compatibility issues with normal bluepill boards/controllers if the binary gets too big.

Maybe checking what the release binary size is compared to the debug release can tell us more about what's eating all the space if the size is the issue.

File: src/bin/gnss7_i2c.rs
Comment (jake-g00dwin) [inline src/bin/gnss7_i2c.rs:1]: Looks good, I'm curious about the frequency though. Is there more information or a documentation line you can add to the source code?

File: src/bin/gnss7_i2c.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    let p = embassy_stm32::init(Default::default());
+    //115200 seems to be recommended
+    let mut i2c = I2c::new(
+        p.I2C2,
+        p.PB10,
+        p.PB11,
+        Irqs,
+        NoDma,
+        NoDma,
+        Hertz(115200),
```
Comment (jake-g00dwin) [inline src/bin/gnss7_i2c.rs:66]: Maybe add a document comment on the reason for the frequency and if it's recommended by the sensor, embassy or someone else?

If possible changing the frequency from a magic number to a constant like:
`pub const MOST_STABLE_NEO9_I2C_FREQ = Hertz(115200);` would be useful to whoever has to work on this later on to give some contextual clues about design choices.

**Review by jake-g00dwin — CHANGES_REQUESTED** [review]
Looks pretty good; couple areas could use more detail. Also maybe consider adding in tests at some point as well; they can double as documentation on the usage of code as well.

File: src/bin/gnss7_i2c.rs (hunk 2 above)
Comment (stackunderfl0w) [reply src/bin/gnss7_i2c.rs:66]: No real reason. Documentation puts max at ~400 khz. 115200 is just a pretty common frequency that was fast and stable.

File: .cargo/config.toml (hunk 1 above)
Comment (stackunderfl0w) [reply .cargo/config.toml:3]: I was just reading the writing on the chips we have. All of mine end in CBT6, might as well have the extra space available if it exists

**Pull request merged**